src/commands/flow/internal/saveToCsv.ts

```diff
@@ -0,0 +1,63 @@
+import { createFile, fileExists, getRealPath, readFile } from '@sasjs/utils'
+import stringify from 'csv-stringify'
+import { displayError } from '../../../utils/displayResult'
+
+// REFACTOR: move to utility
+export const saveToCsv = async (
+  csvFileRealPath: string,
+  data: any,
+  columns: any
+) => {
+  return new Promise(async (resolve, reject) => {
```

Comment:
Because this function is marked as async it will return a Promise anyway, so we don't need to return a Promise explicitly

Reply:
In this routine, we have to wait till async function of `setInterval` to be resolved.